internal/buildinfo/buildinfo.go

```diff
@@ -17,9 +17,30 @@
 package buildinfo
 
 var (
-	// BuildID is the unique build identifier.
-	BuildID string = "unknown"
+	// id is the unique build identifier.
+	id string = "unknown"
 
-	// BuildTag is the git tag from which this build was created.
-	BuildTag string = "unknown"
+	// tag is the git tag from which this build was created.
+	tag string = "unknown"
 )
+
+// Build is an interface to get the build information.
+type Build interface {
+	ID() string
+	Tag() string
+}
+
+// KeyServer provides the build information about the key server.
+type KeyServer struct{}
```

Comment:
Make this private

Reply:
Was this comment on the KeyServer struct, or a previous interface (that I've deleted)?

Reply:
This one - I don't think this should be public. Also see my comment about LDFLAGS and private variables. I don't think you can set `tag` or `id` via an LDFLAG because they are private.

Reply:
Are you saying we should define the struct as private, but have a public variable to use outside of this package?

I did test with setting the private variable using `ldflags` and it does seem to work.